win32 warnings

hi @sreimers 


here are some more win32 warnings with cmake:


```
ninja: warning: multiple rules generate re.lib. builds involving this target will not be correct; continuing anyway [-w dupbuild=warn]

um\qos2.h(39): warning C4142: 'QOS_FLOWID': benign redefinition of type
src\udp\udp.c(35): note: see declaration of 'QOS_FLOWID'
um\qos2.h(39): warning C4142: 'PQOS_FLOWID': benign redefinition of type
src\udp\udp.c(35): note: see declaration of 'PQOS_FLOWID'
```

--- a/src/udp/udp.c
+++ b/src/udp/udp.c
@@ -32,9 +32,11 @@
 #include <re_net.h>
 #include <re_udp.h>
 #ifdef WIN32
+#if !defined(_MSC_VER)
 typedef UINT32 QOS_FLOWID, *PQOS_FLOWID;
 #ifndef QOS_NON_ADAPTIVE_FLOW
 #define QOS_NON_ADAPTIVE_FLOW 0x00000002
+#endif
 #endif
 #include <winsock2.h>
 #include <qos2.h>

